### PR TITLE
doc: Minor fixes in safegcd_implementation.md

### PR DESCRIPTION
**Pull request by robot-dreams**

### CODE REVIEW

File: doc/safegcd_implementation.md [hunk 1]
```diff
@@ -576,7 +576,7 @@ i = N # divsteps left to do
 while True:
     # Get rid of all bottom zeros at once. In the first iteration, g may be odd and the following
     # lines have no effect (until "if eta < 0").
-    zeros = min(i, count_trailing_zeros(g))
+    zeros = i if g == 0 else min(i, count_trailing_zeros(g))
```
Comment (sipa) [inline doc/safegcd_implementation.md:579]: Hmm, it feels like this is mostly due to `count_trailing_zeros` giving the unintuitive *-1* as result when fed *0*.

Maybe we should have `count_trailing_zeros` return *N* on *0*?

Comment (robot-dreams) [reply doc/safegcd_implementation.md]: Yeah, that's much more clear!

File: doc/safegcd_implementation.md [hunk 2]
```diff
@@ -570,7 +570,10 @@ bits efficiently, which is possible on most platforms; it is abstracted here as
 ```python
 def count_trailing_zeros(v):
     """For a non-zero value v, find z such that v=(d<<z) for some odd d."""
```
Comment (sipa) [inline doc/safegcd_implementation.md:572]: The comment would need to be updated.

**Review by real-or-random — APPROVED** [review]
ACK https://github.com/bitcoin-core/secp256k1/pull/1010/commits/dc9b6853b72b9a492cad230623670e89157525ca

**Comment by sipa** [comment]
ACK dc9b6853b72b9a492cad230623670e89157525ca

**Pull request merged**